botocore/session.py

```diff
@@ -972,6 +892,52 @@ def lazy_register_component(self, name, no_arg_factory):
             pass
 
 
+class SessionVarsShim(collections.MutableMapping):
+    def __init__(self, session, *args, **kwargs):
+        self._session = session
+        self._store = dict()
+        self.update(dict(*args, **kwargs))
```

Comment:
This is a little hard to follow what is being passed in here. Is there a reason we do not just accept the original `session_vars_map` dictionary as an optional argument? The nice part about that is we would no longer need an `if` statement when instantiating this.